packages/blade/src/components/Collapsible/__tests__/Collapsible.web.test.tsx

```diff
@@ -0,0 +1,207 @@
+    expect(getByText(bodyText)).not.toBeVisible();
+    expect(getByRole('region', { hidden: true })).toHaveAttribute('aria-hidden', 'true');
+
+    // test built in collapsible button
+    await user.click(collapsibleButton);
+    await waitFor(() => expect(getByText(bodyText)).toBeVisible());
+
+    expect(collapsibleButton).toHaveAttribute('aria-expanded', 'true');
+    expect(getByText(bodyText)).toBeVisible();
+    expect(getByRole('region')).toHaveAttribute('aria-hidden', 'false');
+
+    // test external button as controlled input for collapsible
+    await user.click(externalButton);
+    // getByText seems bugged when collapsing back, even if the container has 0 height and aria-hidden="true" it still shows the text content as visible
+    await waitFor(() => expect(collapsibleButton).toHaveAttribute('aria-expanded', 'false'));
+
+    expect(getByRole('region', { hidden: true })).toHaveAttribute('aria-hidden', 'true');
+  });
+
+  it('should accept testID with CollapsibleLink', () => {
+    const { getByTestId } = renderWithTheme(
+      <Collapsible testID="collapsible">
+        <CollapsibleLink testID="collapsible-link">View Price Breakdown</CollapsibleLink>
+        <CollapsibleBody testID="collapsible-body">
+          <Box display="flex" flexDirection="column" minWidth="200px">
+            <Text>Actual amount</Text>
+            <Amount value={1000} intent="positive" />
+          </Box>
+        </CollapsibleBody>
+      </Collapsible>,
+    );
+
+    getByTestId('collapsible');
+    getByTestId('collapsible-link');
+    getByTestId('collapsible-body');
+  });
+
+  it('should accept testID with CollapsibleButton', () => {
+    const { getByTestId } = renderWithTheme(
+      <Collapsible testID="collapsible">
+        <CollapsibleButton testID="collapsible-button">View Price Breakdown</CollapsibleButton>
+        <CollapsibleBody testID="collapsible-body">
+          <Box display="flex" flexDirection="column" minWidth="200px">
+            <Text>Actual amount</Text>
+            <Amount value={1000} intent="positive" />
+          </Box>
+        </CollapsibleBody>
+      </Collapsible>,
+    );
+
+    getByTestId('collapsible');
+    getByTestId('collapsible-button');
+    getByTestId('collapsible-body');
+  });
+
+  it('should pass general a11y with CollapsibleLink', async () => {
+    const { getByRole } = renderWithTheme(
+      <Collapsible defaultIsExpanded>
+        <CollapsibleLink>View Price Breakdown</CollapsibleLink>
+        <CollapsibleBody>
+          <Box display="flex" flexDirection="column" minWidth="200px">
+            <Text>Actual amount</Text>
+            <Amount value={1000} intent="positive" />
+          </Box>
+        </CollapsibleBody>
+      </Collapsible>,
+    );
+
+    const collapsibleTrigger = getByRole('button');
+    const collapsiblePanel = getByRole('region');
+
+    await assertAccessible(collapsibleTrigger);
+    await assertAccessible(collapsiblePanel);
```

Comment:
same here, assert whole tree.

Reply:
Done in https://github.com/razorpay/blade/pull/1360/commits/5362504ae49ec5bbbae577c5543de2e9ba15b3cc